src/lib.rs

```diff
@@ -149,7 +149,23 @@ impl Key for &str {
 
 /// Trait exposing serialization/deserialization of nodes by keys/paths and traversal.
 ///
-/// The depth parameter `Y` is the maximum key length: the depth/height of the tree.
+/// The depth parameter `Y` is the maximum miniconf recursion depth.
+/// An implementor chooses a `Z <= Y` and may then in the recursive methods (directly
+/// and indirectly) retrieve at most `Z` keys from the `keys` and `indices` iterators
+/// and call the `func` callback at most `Z` times. It may
```

Comment:
I don't understand these docs - what `keys` and `indices` are they referring to? Seems like `indices` is referring to the member on this trait, but `keys` might be an arg provided? In general, I'm having a hard time understanding these docs and what they're supposed to mean

Reply:
Clarified the docs. Please check again.